rust/src/stracciatella.rs

```diff
@@ -288,6 +289,89 @@ pub unsafe extern "C" fn guess_resource_version(
     return result;
 }
 
+/// Returns the path to game.json.
+#[no_mangle]
+pub unsafe extern "C" fn get_game_json_path() -> *mut c_char {
+    let mut path = PathBuf::new();
+    let extra_data_dir = option_env!("EXTRA_DATA_DIR");
+    if extra_data_dir.is_some() && extra_data_dir.unwrap().len() > 0 {
+        // use dir defined at compile time
+        path.push(extra_data_dir.unwrap());
+    } else if let Ok(exe) = env::current_exe() {
+        if let Some(dir) = exe.parent() {
+            // use the directory of the executable
+            path.push(dir);
+        }
+    }
+    path.push("externalized/game.json");
+    let path: String = path.to_string_lossy().into();
+    return CString::new(path).unwrap().into_raw();
+}
+
+/// Finds a path relative to the stracciatella home directory.
+/// If path is null, it finds the stracciatella home directory.
+/// If test_exists is true, it makes sure the path exists.
+#[no_mangle]
+pub extern "C" fn find_path_from_stracciatella_home(
+    path: *const c_char,
+    test_exists: bool,
+) -> *mut c_char {
+    if let Ok(mut path_buf) = find_stracciatella_home() {
+        if !path.is_null() {
+            let c_s = unsafe { CStr::from_ptr(unsafe_from_ptr!(path)) };
+            let s = String::from_utf8_lossy(c_s.to_bytes());
+            path_buf.push(s.as_ref());
+        }
+        if test_exists && !path_buf.exists() {
+            return ptr::null_mut(); // path not found
+        } else {
+            match path_buf.canonicalize() {
+                Ok(p) => path_buf = p,
+                _ => {}
+            }
+            let s: String = path_buf.to_string_lossy().into();
+            return CString::new(s).unwrap().into_raw(); // path found
+        }
+    } else {
+        return ptr::null_mut(); // no home
+    }
+}
+
+/// Returns true if it was able to find path relative to base.
+/// Makes caseless searches one component at a time.
+#[no_mangle]
+pub extern "C" fn check_if_relative_path_exists(
```

Comment:
I think this needs to be case insensitive. For me the warnings always show.

Reply:
I'll look into it.

Reply:
Fixed.